### ISSUE TITLE
askama loops with actix-web - cannot move out of shared reference

### ISSUE THREAD

**Issue by tungli** [actix/actix-web#2030]
I am struggling to get `askama` template for-loops working with actix-web.
Specifically, taking [this example](https://github.com/actix/examples/tree/master/template_engines/askama) and changing the `index.html` template to this:

```
<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8" />
  <title>Actix web</title>
</head>
<body>
  <p>
    {% for i in 0..100 %}
     {{ i }}
    {% endfor %}
  </p>
</body>
</html>
```

I get:

```
error[E0507]: cannot move out of a shared reference
  --> template_askama/src/main.rs:13:10
   |
13 | #[derive(Template)]
   |          ^^^^^^^^ move occurs because value has type `std::ops::Range<i32>`, which does not implement the `Copy` trait
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0507`.
error: could not compile `template-askama`.
```

which is a bit puzzling since the template does not reference any data and without actix-web this works.

Can anybody clarify how this happens, please?

**Comment by JohnTitor** [comment]
What else did you change?
From the error, it doesn't look like actix-web causes the problem.

**Comment by tungli** [comment]
Nothing.

Any for loop through data that are not directly the data in the template struct does not work.
This includes `zip`s, `enumerate`, ranges..

Should I report this to the askama repo, instead?

**Comment by JohnTitor** [comment]
Just checked with your index.html but it just works fine on my end. I doubt it's even an issue on askama, I guess some templates and/or code on https://github.com/djc/askama/tree/main/testing may help (and note that our example uses an old version of askama).

**Comment by tungli** [comment]
Ok I messed up a bit, but now I am even more confused, because:

1. @JohnTitor you are right -> the example I posted does indeed work. Sorry about that!
2. [This](https://github.com/tungli/examples/blob/master/template_engines/askama/templates/index.html), however does not compile, not even with the latest askama version.
3. I must have removed the parenthesis during editing of the issue, because this also does not compile:
```
<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8" />
  <title>Actix web</title>
</head>
<body>
  <p>
    {% for i in (0..100) %}
     {{ i }}
    {% endfor %}
  </p>
</body>
</html>
```

I also added the examples to [tests](https://github.com/tungli/askama/blob/main/testing/tests/this_works.rs) to askama to see that this is not an askama problem (i.e these compile and run fine).

**Comment by JohnTitor** [comment]
Seems you're using the code on master, not the 0.9 tag. I think this issue is related to https://github.com/djc/askama/issues/107, i.e., it's fixed on master (or even on 0.10, I'm not sure though) but it still exists on 0.9 thus it doesn't compile. So, I think this isn't an issue about actix-web at all.
